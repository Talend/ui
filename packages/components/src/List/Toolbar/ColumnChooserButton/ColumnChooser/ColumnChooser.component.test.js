/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { mount } from 'enzyme';
import Component from './ColumnChooser.component';

const columns = [
	{ key: 'id', label: 'Id', order: 1 },
	{ key: 'name', label: 'Name', order: 2 },
	{ key: 'author', label: 'Author', order: 3 },
	{ key: 'created', label: 'Created', order: 6 },
	{
		key: 'modified',
		label: 'Very long name long name long name long name long name',
		order: 4,
		header: 'icon',
		data: { iconName: 'talend-scheduler' },
	},
	{ key: 'icon', label: 'Icon', hidden: true, order: 5 },
];

describe('ColumnChooser', () => {
	it('should render with default props', () => {
		// Given
		const props = {
			id: 'my-id',
			columnsFromList: columns,
			onSubmit: jest.fn(),
		};
		// When
		const wrapper = mount(<Component {...props} />);
		// Then
		expect(wrapper.html()).toMatchSnapshot();
	});
	it('should render with children', () => {
		// Given
		const props = {
			id: 'my-id',
			columnsFromList: columns,
			onSubmit: jest.fn(),
		};
		const Children = <div id="my-child">Hello World</div>;
		// When
		const wrapper = mount(<Component {...props}>{Children}</Component>);
		// Then
		expect(wrapper.find('div#my-child').text()).toBe('Hello World');
	});
	it('should trigger the onSubmit props', () => {
		// Given
		const onSubmit = jest.fn();
		const props = {
			id: 'my-id',
			columnsFromList: columns,
			onSubmit,
		};
		// When
		const wrapper = mount(<Component {...props} />);
		wrapper.find('form#my-id-form').simulate('submit');
		// Then
		expect(onSubmit).toHaveBeenCalled();
		expect(onSubmit.mock.calls[0][1]).toEqual([
			{ hidden: false, label: 'Id', order: 1 },
			{ hidden: false, label: 'Name', order: 2 },
			{ hidden: false, label: 'Author', order: 3 },
			{ hidden: false, label: 'Very long name long name long name long name long name', order: 4 },
			{ hidden: true, label: 'Icon', order: 5 },
			{ hidden: false, label: 'Created', order: 6 },
		]);
	});
	it('should filter the columns by name if an initial filter value is provided', () => {
		// Given
		const props = {
			id: 'my-id',
			initialFilterValue: 'Name',
			columnsFromList: columns,
			onSubmit: jest.fn(),
		};
		// When
		const wrapper = mount(<Component {...props} />);
		// Then
		expect(
			wrapper.find('input[aria-describedby="my-id-body-display the column Author"]'),
		).toHaveLength(0);
		expect(
			wrapper.find('input[aria-describedby="my-id-body-display the column Name"]'),
		).toHaveLength(1);
	});
	it('should locked the first two columns', () => {
		// Given
		const props = {
			id: 'my-id',
			columnsFromList: columns,
			onSubmit: jest.fn(),
			nbLockedLeftItems: 2,
		};
		// When
		const wrapper = mount(<Component {...props} />);
		// Then
		expect(wrapper.find('svg[name="talend-locked"]')).toHaveLength(2);
	});
});
