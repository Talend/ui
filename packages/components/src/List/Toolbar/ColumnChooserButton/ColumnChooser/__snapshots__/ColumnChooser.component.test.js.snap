// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChooser should render with default props 1`] = `
<form id="my-id-form"
      class="tc-column-chooser theme-tc-column-chooser"
>
  <div id="column-chooser-body"
       class="theme-body tc-tooltip-body"
  >
    <div id="column-chooser-body-content"
         class="theme-content tc-tooltip-content"
    >
      <div class="tc-column-chooser-body theme-tc-column-chooser-body">
        <div class="tc-column-chooser-row theme-tc-column-chooser-row tc-column-chooser-row-select-all theme-tc-column-chooser-row-select-all theme-tc-column-chooser-row-select-all theme-tc-column-chooser-row-select-all">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Columns"
                   data-feature="column-chooser.select.all.enable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Columns"
                     data-checked="0"
                     aria-describedby="my-id-body-display all the columns"
              >
              <span>
                Columns
              </span>
            </label>
          </div>
          <div id="my-id-body-display all the columns"
               class="sr-only"
          >
            display all the columns
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Id"
                   data-feature="column-chooser.select.disable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Id"
                     data-checked="2"
                     aria-describedby="my-id-body-display the column Id"
                     checked
              >
              <span>
                Id
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Id"
               class="sr-only"
          >
            display the column Id
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Name"
                   data-feature="column-chooser.select.disable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Name"
                     data-checked="2"
                     aria-describedby="my-id-body-display the column Name"
                     checked
              >
              <span>
                Name
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Name"
               class="sr-only"
          >
            display the column Name
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Author"
                   data-feature="column-chooser.select.disable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Author"
                     data-checked="2"
                     aria-describedby="my-id-body-display the column Author"
                     checked
              >
              <span>
                Author
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Author"
               class="sr-only"
          >
            display the column Author
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Very long name long name long name long name long name"
                   data-feature="column-chooser.select.disable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Very long name long name long name long name long name"
                     data-checked="2"
                     aria-describedby="my-id-body-display the column Very long name long name long name long name long name"
                     checked
              >
              <span>
                Very long name long name long name long name long name
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Very long name long name long name long name long name"
               class="sr-only"
          >
            display the column Very long name long name long name long name long name
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Icon"
                   data-feature="column-chooser.select.enable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Icon"
                     data-checked="0"
                     aria-describedby="my-id-body-display the column Icon"
              >
              <span>
                Icon
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Icon"
               class="sr-only"
          >
            display the column Icon
          </div>
        </div>
        <div class="tc-column-chooser-row theme-tc-column-chooser-row">
          <div class="checkbox tc-toggle theme-tc-toggle checkbox">
            <label for="my-id-body-checkbox-Created"
                   data-feature="column-chooser.select.disable"
            >
              <input type="checkbox"
                     id="my-id-body-checkbox-Created"
                     data-checked="2"
                     aria-describedby="my-id-body-display the column Created"
                     checked
              >
              <span>
                Created
              </span>
            </label>
          </div>
          <div id="my-id-body-display the column Created"
               class="sr-only"
          >
            display the column Created
          </div>
        </div>
      </div>
    </div>
  </div>
  <footer id="column-chooser-footer"
          class="theme-footer tc-tooltip-footer tc-column-chooser-footer theme-tc-column-chooser-footer"
  >
    <button aria-label="Apply"
            id="my-id-submit-button"
            type="submit"
            class="btn btn-info"
    >
      <span>
        Apply
      </span>
    </button>
  </footer>
</form>
`;
